addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -738,7 +734,10 @@ export class Orderline extends PosModel {
             orderline.compute_fixed_price(order_line_price),
             this.pos.currency.decimal_places
         );
-        // only orderlines of the same product can be merged
+        let hasSameAttributes = true;
+        if(Object(orderline.attribute_value_ids).length && Object(this.attribute_value_ids).length) {
```

Comment:
if you have one orderline that has attributes and then another that doesn't have attributes this will not go into the `if` and then `hasSameAttributes` will still be `true`, right?

Reply:
what do you think about this approach?
```js
const hasSameAttribues = orderline.attribute_value_ids?.length === this.attribute_value_ids?.length

```